crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -325,6 +324,57 @@ impl CumulocityConverter {
         })
     }
 
+    pub async fn try_register_source_entities(
+        &mut self,
+        message: &MqttMessage,
+    ) -> Result<Vec<PendingEntityData>, ConversionError> {
+        if let Ok((source, channel)) = self.mqtt_schema.entity_channel_of(&message.topic) {
+            match channel {
+                Channel::EntityMetadata => {
+                    if let Ok(register_message) = EntityRegistrationMessage::try_from(message) {
+                        return self
+                            .try_register_entity_with_pending_children(register_message)
+                            .await;
+                    }
+                    Err(anyhow!(
+                        "Invalid entity registration message received on topic: {}",
+                        message.topic.name
+                    )
+                    .into())
+                }
+                _ => {
+                    // if the target entity is unregistered, try to register it first using auto-registration
+                    if self.entity_store.get(&source).is_none()
+                        && self.config.enable_auto_register
+                        && source.matches_default_topic_scheme()
```

Comment:
I would be better to make these checks inside `self.try_auto_register_entity(&source)`.